ukis_pysat/download.py

```diff
@@ -382,6 +346,156 @@ def __exit__(self, exc_type, exc_val, exc_tb):
+    size: str = field(check_type=True, default="", doc="File size [MB]")
+    srcid: str = field(check_type=True, doc="Source product ID")
+    srcurl: str = field(check_type=True, default="", doc="Source product URL")
+    srcuuid: str = field(check_type=True, doc="Source product UUID")
+    geom: dict = field(check_type=True, default=None, doc="Geometry [multipolygon dict]")
+
+    @acquisitiondate.converter(accepts=str)
+    @ingestiondate.converter(accepts=str)
+    @processingdate.converter(accepts=str)
+    def _prep_date(self, value):
+        """Converts a date string to datetime.date object.
+
+        :returns: Datetime.date
+        """
+        if value is not None:
+            return parse(value)
+
+    def to_dict(self):
+        """Converts Metadata to Dict.
+
+        :returns: Metadata (Dict)
+        """
+        return {
+            "id": self.id,
+            "platformname": None if self.platformname is None else self.platformname.value,
+            "producttype": self.producttype,
+            "orbitdirection": self.orbitdirection,
+            "orbitnumber": self.orbitnumber,
+            "relativeorbitnumber": self.relativeorbitnumber,
+            "acquisitiondate": None if self.acquisitiondate is None else self.acquisitiondate.strftime("%Y/%m/%d"),
+            "ingestiondate": None if self.ingestiondate is None else self.ingestiondate.strftime("%Y/%m/%d"),
+            "processingdate": None if self.processingdate is None else self.processingdate.strftime("%Y/%m/%d"),
+            "processingsteps": self.processingsteps,
+            "processingversion": self.processingversion,
+            "bandlist": self.bandlist,
+            "cloudcoverpercentage": self.cloudcoverpercentage,
+            "format": self.format,
+            "size": self.size,
+            "srcid": self.srcid,
+            "srcurl": self.srcurl,
+            "srcuuid": self.srcuuid,
+        }
+
+    def to_json(self):
+        """Converts Metadata to JSON.
+
+        :returns: Metadata (JSON)
+        """
+        return json.dumps(self.to_dict())
+
+    def to_geojson(self):
+        """Converts Metadata to GeoJSON.
+
+        :returns: Metadata (GeoJSON)
+        """
+        return geometry.mapping(_GeoInterface({"type": "Feature", "properties": self.to_dict(), "geometry": self.geom}))
+
+    def save(self, target_dir):
+        """Saves Metadata to GeoJSON file in target_dir with srcid as filename.
+
+        :param target_dir: Target directory that holds the downloaded metadata (String)
+        """
+        g = self.to_geojson()
+        with open(os.path.join(target_dir, g["properties"]["srcid"] + ".json"), "w") as f:
+            json.dump(g, f)
+
+
+class MetadataCollection:
+    """
+    Provides a container to store a collection of metadata objects. Conversion methods are provided to
+    analyse the metadata collection further.
+    """
+
+    __init__ = make_init()
+    items: List[Metadata] = field(type_hint=List[Metadata])

```

Comment:
Shouldn't this type alias be alist already? I don't really get this, but could also be me.

Reply:
Need to check this one again. I think you are right, now that I look at it again.

Reply:
We'll keep it for the merge and will check what to do with this later on.